### PR TITLE
Update Yarn and Node

### PR DESCRIPTION
**Pull request by scinos**
# Pull Request Template for Home Assistant / Lovelace Card Repository

## Overview

This PRs updates Yarn to a non-deprecated versions, and Node to the latest LTS version. This should make development 
easier as we can use Yarn from Node's corepack, no need to try complex methods to install Yarn 1.x

### Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Documentation update
- [ ] Refactoring
- [ ] Other (please describe):

### CODE REVIEW

File: .eslintrc.cjs
Comment (scinos) [inline .eslintrc.cjs:8]: `lines-between-class-members` got [moved](https://typescript-eslint.io/rules/lines-between-class-members/) to `@stylstic/js`.

File: .eslintrc.cjs
Comment (scinos) [inline .eslintrc.cjs:83]: Before this change, ESLint was failing for test files. It should work now.

File: .gitattributes
Comment (scinos) [inline .gitattributes:1]: Yarn does this automatically when migrating from Yarn 1.x 🤷🏼

File: package.json
Comment (scinos) [inline package.json:10]: Modern yarn has its own shell, to guarantee that things like `rm -fr` works the same across all platforms, including Windows.

**Comment by ThomDietrich** [comment]
Hey @scinos,
thanks for your support! This PR to update dependencies was desperately needed :) I myself am just a repo maintainer and have no js/node experience.

If you are interested, could you also check https://github.com/rejuvenate/lovelace-horizon-card/pull/131 and see if further additions in your PR (or another) would be worthwhile?

Best!

**Comment by scinos** [comment]
> This PR to update dependencies was desperately needed :)

I'm happy to update this lib and bring it to modern standards :) This PR is still in draft because I broke many tests, I'd like to fix them before merging it.

> If you are interested, could you also check https://github.com/rejuvenate/lovelace-horizon-card/pull/131 and see if further additions in your PR (or another) would be worthwhile?

Sure!

**Comment by scinos** [comment]
PR ready for review. I managed to update Yarn+Node without having to change ESLint or Typescript, so this PR is now smaller than it was when I initially open it.

I can't see CI results until this PR is approved, but locally everything seems to work:

Screenshot 2024-11-05 at 10 35 03

**Review by ThomDietrich — APPROVED** [review]

**Comment by ThomDietrich** [comment]
Thanks! I've merged the PR to have a baseline CI response. The Build and Release job errors out with "Run actions/cache@v1: Error: Input required and not supplied: path"

**Pull request merged**
